### ISSUE TITLE
support .var for large durations

### ISSUE THREAD

**Issue by rancomp** [pola-rs/polars#12896]
### Description

`.std` and `.var` are not supported for Series and Expressions of `DataType::Duration`.

When calculating the variance, the calculation sometimes fails to handle large integers that arise due to the representation of durations in microseconds. This issue seems to be leading to overflows, resulting in computation faults.

For example, 
```
>>>pl.DataFrame(
    {"c":[timedelta(days=1), timedelta(days=2), timedelta(days=4)]}
).select(var=pl.var("c"), std=pl.std("c"))
shape: (1, 2)
┌──────────────┬─────────────────────────┐
│ var          ┆ std                     │
│ ---          ┆ ---                     │
│ duration[μs] ┆ duration[μs]            │
╞══════════════╪═════════════════════════╡
│ null         ┆ 1d 12h 39m 38s 180014µs │
└──────────────┴─────────────────────────┘
```
The variance should be `~2d 8h`, but it's failing because it's carrying very large integers and casting them from `DataType::Float64` to integers.

My naive approach was to overcome this through:
`1. normalizing by the maximum duration, 2. calculating variance, 3. multiplying back with the square.`
But alas, the casting still fails. I'm sure there are better representations of durations during this calculation. This may impact precision, but all in all it's still preferable to returning `null`.

I'm asking for suggestions on how to handle this _simple_ calculation. I'm guessing there's a way but, I'm not familiar with the Rust ecosystem.

**Comment by ion-elgreco** [comment]
Related PR https://github.com/pola-rs/polars/pull/12881 where there is a short discussion on this

**Comment by ion-elgreco** [comment]
@rancomp how would you get to variance 2d 8h?

**Comment by rancomp** [comment]
Thanks @ion-elgreco , yeah I saw the discussion, just thought this point deserves an issue in its own right.
Regarding the variance, I was calculating the sample variance, `sample_variance = sum_of_squares / n-1`. I now edited the original message to reflect this.

**Comment by mcrumiller** [comment]
@rancomp I can put this into my PR where I'll be fixing the dtype returns for `std` and `var` for temporals.

**Comment by ion-elgreco** [comment]
@mcrumiller what's the return type going to be for std?

**Comment by mcrumiller** [comment]
All will return a duration (`pl.Duration` or `datetime.timedelta`).

**Comment by mcrumiller** [comment]
They all return null for `std` and `var`:

```python
import polars as pl
from datetime import date, datetime, time, timedelta

pl.Series([date(2023, 1, 1), date(2023, 1, 2)]).std() # returns null
pl.Series([datetime(2023, 1, 1), datetime(2023, 1, 2)]).std() # returns null
pl.Series([timedelta(2023, 1, 1), timedelta(2023, 1, 2)]).std() # returns null
pl.Series([time(1, 1, 1), time(1, 1, 2)]).std() # returns null
```

**Comment by mcrumiller** [comment]
@ion-elgreco I've got a working version ready to go, just waiting for other PR first.

I opted to convert to Float64 and represent the duration as number of days, then perform the var/std, then convert back. We can increase the precision by increasing the units of representation (i.e. for `pl.Time` if we'd could probably get away with units of seconds instead, since `pl.Time` ranges up to 1 day anyway.

**Comment by ion-elgreco** [comment]
> @ion-elgreco I've got a working version ready to go, just waiting for other PR first.
> 
> I opted to convert to Float64 and represent the duration as number of days, then perform the var/std, then convert back. We can increase the precision by increasing the units of representation (i.e. for `pl.Time` if we'd could probably get away with units of seconds instead, since `pl.Time` ranges up to 1 day anyway.

Hmm but for std you will lose quite some precision there, I think for std you can safely do without representing numbers for days.

**Comment by mcrumiller** [comment]
@ion-elgreco I agree with you, and I think for `datetime` durations I'll normalize by 1s (as a starting point) and not at all for `time`s, since there are only 1e9 unique time points anyway, so I don't anticipate running into the issue.

The `std` calculation is simply the square root of the variance, and both require computing the sum-of-squares.
